p4include/v1model.p4

```diff
@@ -140,6 +135,27 @@ extern void clone3<T>(in CloneType type, in bit<32> session, in T data);
 
 extern void truncate(in bit<32> length);
 
+/**
+Verifies the 16-bit checksum of the supplied data.
+T must be a list expression where all the fields are bit-fields or varbits
+and where the total dynamic length of the fields is a multiple of 16 bits.
+If this method detects that a checksum of the data is not correct it
+sets an internal error flag.
```

Comment:
isn't the plan to expose this information as standard_metadata which can be consumed in ingress?